aten/src/ATen/FunctionalStorageImpl.h

```diff
@@ -65,8 +65,8 @@ struct ViewMeta {
 class Alias {
   public:
     struct Update {
-        const at::Tensor& new_val;
-        const std::vector<ViewMeta>& view_metas;
+        const at::Tensor new_val;
+        const std::vector<ViewMeta> view_metas;
```

Comment:
I'm not concerned about the Tensor copy (because that's just a refcount bump). The vector copy has the potential to be bad for performance, especially if the vector is very large.

One way to do this is to give FunctionalTensorWrapper a `shared_ptr<std::vector<ViewMeta>>`. You'd want to benchmark this though because it's another layer of indirection.

Another alternative is to, instead of using `vector<ViewMeta>`, use `SmallVector<N, ViewMeta>`, where N is some number. I don't see people chain more than 5 views in a row in models usually, so N=5 seems like a reasonable threshold.

Add to the followup list maybe? (Btw, do you have a list of planned follow ups somewhere?)

Reply:
I don't have a github issue of followups open - I'm just tracking them locally for now. But I should add a github issue for them soon. Just to lay them out here (will add them on a github issue later):

correctness:
- handling scalars properly
- handling ` List[Optional[Tensor]]` arguments properly
- handling nested functionalize transforms
- adding OpInfo testing

non-functorch use cases:
- split up the pass across 3 separate dispatch keys

performance (there's some low-hanging perf fruit that's probably worth doing before using this in eager mode backends like vulkan/XLA):
- Use `SmallVector<ViewMeta>` (and consider benchmarking against `shared_ptr<vector<ViewMeta>>`)
- remove the boxed fallback and use codegen'd kernels instead for non-view / non-inplace ops